packages/server/src/course/course.service.ts

```diff
@@ -118,6 +120,86 @@ export class CourseService {
     }
   }
 
+  async editCourse(
+    courseId: number,
+    coursePatch: EditCourseInfoParams,
+  ): Promise<void> {
+    const course = await CourseModel.findOne(courseId);
+    if (course === null || course === undefined) {
+      throw new HttpException(
+        ERROR_MESSAGES.courseController.courseNotFound,
+        HttpStatus.NOT_FOUND,
+      );
+    }
+
+    if (Object.values(coursePatch).some((x) => x === null || x === '')) {
+      throw new BadRequestException(
+        ERROR_MESSAGES.courseController.updateCourse,
+      );
+    }
+
+    for (const crn of new Set(coursePatch.crns)) {
+      const courseCrnMap = await CourseSectionMappingModel.findOne({
+        crn: crn,
+      });
+
+      let conflictCourse;
+      if (courseCrnMap) {
+        conflictCourse = await CourseModel.findOne(courseCrnMap.courseId);
+        if (
+          courseCrnMap.courseId !== courseId &&
+          conflictCourse &&
+          conflictCourse.semesterId === course.semesterId
+        ) {
+          throw new BadRequestException(
+            ERROR_MESSAGES.courseController.crnAlreadyRegistered(crn, courseId),
+          );
+        }
+      }
+      if (
+        !courseCrnMap ||
```

Comment:
Do you need the logic for the second half of this if statement if you already check and throw the exception above?

Reply:
yes 🐻‍❄️